handsontable/src/3rdparty/walkontable/src/utils/stylesHandler.js

```diff
@@ -0,0 +1,188 @@
+const CLASSIC_THEME_DEFAULT_HEIGHT = 23;
+
+/**
+ * Handles the theme-related style operations.
+ */
+export class StylesHandler {
+  /**
+   * The instance's root element.
+   *
+   * @type {HTMLElement}
+   */
+  #rootElement;
+
+  /**
+   * The computed style of the root element.
+   *
+   * @type {CSSStyleDeclaration}
+   * @private
+   */
+  #rootComputedStyle;
+
+  /**
+   * The root document of the instance.
+   *
+   * @type {Document}
+   * @private
+   */
+  #rootDocument;
+
+  /**
+   * `true` if the classic theme is enabled, `false` otherwise.
+   *
+   * @type {boolean}
+   */
+  #isClassicTheme = false;
+
+  /**
+   * An object to store CSS variable values.
+   *
+   * @type {object}
+   * @private
+   */
+  #cssVars = {};
+
+  /**
+   * Stores the computed styles for various elements.
+   *
+   * @type {object} - An object containing the computed styles if a nested structore of `element: { [element type]: {property: value} }`.
+   * @private
+   */
+  #computedStyles = {};
+
+  /**
+   * Initializes a new instance of the `StylesHandler` class.
+   *
+   * @param {object} domBindings - The DOM bindings for the instance.
+   */
+  constructor(domBindings) {
+    this.#rootElement = domBindings.rootTable.parentElement;
+    this.#rootComputedStyle = getComputedStyle(this.#rootElement);
+    this.#rootDocument = domBindings.rootDocument;
+
+    // TODO: naive method of checking for modern themes
+    if (this.#rootComputedStyle?.getPropertyValue('--ht-line-height') === '') {
```

Comment:
The computed style always exists. I guess it's not necessary to use optional chaining.
```suggestion
    if (this.#rootComputedStyle.getPropertyValue('--ht-line-height') === '') {
```